nemo_aligner/data/nlp/builders.py

```diff
@@ -262,38 +262,20 @@ def build_dataset(index, name):
 
 def build_sft_dataset(data_cfg, tokenizer, num_samples, answer_only_loss=True, is_chat=True, special_tokens=None):
     dataset_cls = GPTSFTChatDataset if is_chat else GPTSFTDataset
-    dataset = dataset_cls(
-        file_path=data_cfg.file_path,
-        tokenizer=tokenizer,
-        max_seq_length=data_cfg.max_seq_length,
-        min_seq_length=data_cfg.min_seq_length,
-        add_bos=data_cfg.get("add_bos", False),
-        add_eos=data_cfg.get("add_eos", True),
-        add_sep=data_cfg.get("add_sep", False),
-        sep_id=0,
-        max_num_samples=num_samples,
-        seed=data_cfg.get("seed", 1234),
-        label_key=data_cfg.get("label_key", "answer"),
-        answer_only_loss=answer_only_loss,
-        truncation_field=data_cfg.get("truncation_field", "text"),
-        pad_to_max_length=data_cfg.get("pad_to_max_length", False),
-        index_mapping_dir=data_cfg.get("index_mapping_dir", None),
-        prompt_template=data_cfg.get("prompt_template", None),
-        virtual_tokens=0,
-        tokens_to_generate=data_cfg.get(
-            "tokens_to_generate", 0
-        ),  # used at inference time to allocate tensor positions for tokens that will be generated by inf procedure.
-        memmap_workers=data_cfg.get(
-            "memmap_workers", None
-        ),  # used to set num. of workers to create the memmap index files
-        hf_dataset=data_cfg.get(
-            "hf_dataset", False
-        ),  # Whether to load the json file with the HuggingFace dataset. otherwise, will load the jsonl file with the JSONLMemMapDataset.
-        truncation_method=data_cfg.get(
-            "truncation_method", "right"
-        ),  # used to choose truncation method. Options: ['random', 'left', 'right']
-        special_tokens=special_tokens,
-    )
+    # parameters that don't come from the config
+    dataset_params = {
+        "tokenizer": tokenizer,
+        "answer_only_loss": answer_only_loss,
+        "special_tokens": special_tokens,
+        "max_num_samples": num_samples,
+        "sep_id": 0,
+        "virtual_tokens": 0,
+    }
+    dataset_params.update(data_cfg)
+    # parameters that are in the config, but not in the dataset's init
+    for param_name in ["global_batch_size", "micro_batch_size", "shuffle", "drop_last"]:
```

Comment:
It may be safer to specify which parameters are in the dataset class rather than which ones aren't. This is because we don't have much control on the input config and sometimes we inherit a bunch of stuff from the way the base model was trained, that we don't care about.